crates/ironrdp-rdpdr/src/pdu/efs.rs

```diff
@@ -968,57 +970,60 @@ impl ServerDeviceAnnounceResponse {
 /// This enum includes some basic ones for communicating with the RDP server.
 ///
 /// [2.3.1 NTSTATUS Values]: https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-erref/596a1078-e883-4972-9bbc-49e60bebca55
-#[derive(Debug, PartialEq, Copy, Clone)]
-#[repr(u32)]
-pub enum NtStatus {
-    /// STATUS_SUCCESS
-    Success = 0x0000_0000,
-    /// STATUS_UNSUCCESSFUL
-    Unsuccessful = 0xC000_0001,
-    /// STATUS_NOT_IMPLEMENTED
-    NotImplemented = 0xC000_0002,
-    /// STATUS_NO_MORE_FILES
-    NoMoreFiles = 0x8000_0006,
-    /// STATUS_OBJECT_NAME_COLLISION
-    ObjectNameCollision = 0xC000_0035,
-    /// STATUS_ACCESS_DENIED
-    AccessDenied = 0xC000_0022,
-    /// STATUS_NOT_A_DIRECTORY
-    NotADirectory = 0xC000_0103,
-    /// STATUS_NO_SUCH_FILE
-    NoSuchFile = 0xC000_000F,
-    /// STATUS_NOT_SUPPORTED
-    NotSupported = 0xC000_00BB,
-    /// STATUS_DIRECTORY_NOT_EMPTY
-    DirectoryNotEmpty = 0xC000_0101,
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct NtStatus(u32);
+
+impl NtStatus {
+    pub const SUCCESS: Self = Self(0x0000_0000);
+    pub const UNSUCCESSFUL: Self = Self(0xC000_0001);
+    pub const NOT_IMPLEMENTED: Self = Self(0xC000_0002);
+    pub const NO_MORE_FILES: Self = Self(0x8000_0006);
+    pub const OBJECT_NAME_COLLISION: Self = Self(0xC000_0035);
+    pub const ACCESS_DENIED: Self = Self(0xC000_0022);
+    pub const NOT_A_DIRECTORY: Self = Self(0xC000_0103);
+    pub const NO_SUCH_FILE: Self = Self(0xC000_000F);
+    pub const NOT_SUPPORTED: Self = Self(0xC000_00BB);
+    pub const DIRECTORY_NOT_EMPTY: Self = Self(0xC000_0101);
```

Comment:
I guess we can still keep the full name as docstrings for cross-referencing?
```suggestion
    /// STATUS_SUCCESS
    pub const SUCCESS: Self = Self(0x0000_0000);
    /// STATUS_UNSUCCESSFUL
    pub const UNSUCCESSFUL: Self = Self(0xC000_0001);
    /// STATUS_NOT_IMPLEMENTED
    pub const NOT_IMPLEMENTED: Self = Self(0xC000_0002);
    /// STATUS_NO_MORE_FILES
    pub const NO_MORE_FILES: Self = Self(0x8000_0006);
    /// STATUS_OBJECT_NAME_COLLISION
    pub const OBJECT_NAME_COLLISION: Self = Self(0xC000_0035);
    /// STATUS_ACCESS_DENIED
    pub const ACCESS_DENIED: Self = Self(0xC000_0022);
    /// STATUS_NOT_A_DIRECTORY
    pub const NOT_A_DIRECTORY: Self = Self(0xC000_0103);
    /// STATUS_NO_SUCH_FILE
    pub const NO_SUCH_FILE: Self = Self(0xC000_000F);
    /// STATUS_NOT_SUPPORTED
    pub const NOT_SUPPORTED: Self = Self(0xC000_00BB);
    /// STATUS_DIRECTORY_NOT_EMPTY
    pub const DIRECTORY_NOT_EMPTY: Self = Self(0xC000_0101);
```